OpenRA.Mods.Common/Effects/RevealShroudEffect.cs

```diff
@@ -0,0 +1,90 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2016 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Effects;
+using OpenRA.Graphics;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Effects
+{
+	public class RevealShroudEffect : IEffect
+	{
+		static readonly PPos[] NoCells = { };
+
+		WPos pos;
+		Player player;
+		Shroud.SourceType sourceType;
+		int remainingTicks;
+		int remainingDelay;
+		WDist revealRadius;
+		Stance validStances;
+
+		bool cellsRevealed;
+
+		public RevealShroudEffect(WPos pos, WDist radius, Shroud.SourceType type, Player forPlayer, Stance stances, int delay = 0, int duration = 50)
+		{
+			this.pos = pos;
+			player = forPlayer;
+			remainingTicks = duration;
+			remainingDelay = delay;
+			revealRadius = radius;
+			validStances = stances;
+			sourceType = type;
+		}
+
+		void AddCellsToPlayerShroud(Player p, PPos[] uv)
```

Comment:
Please move these two inline, there's no real benefit to having them split out in this case.